src/codegen/tests/snapshots/rusty__codegen__tests__code_gen_tests__array_of_struct_as_member_of_another_struct_and_variable_declaration_is_initialized.snap

```diff
@@ -0,0 +1,22 @@
+---
+source: src/codegen/tests/code_gen_tests.rs
+expression: res
+---
+; ModuleID = 'main'
+source_filename = "main"
+
+%mainProg = type { [5 x %STRUCT1] }
+%STRUCT1 = type { i16, [5 x %STRUCT2] }
+%STRUCT2 = type { i8, i32 }
+
+@mainProg_instance = global %mainProg { [5 x %STRUCT1] [%STRUCT1 { i16 1, [5 x %STRUCT2] [%STRUCT2 { i8 1, i32 128 }, %STRUCT2 { i8 0, i32 1024 }, %STRUCT2 zeroinitializer, %STRUCT2 zeroinitializer, %STRUCT2 zeroinitializer] }, %STRUCT1 { i16 2, [5 x %STRUCT2] [%STRUCT2 { i8 1, i32 256 }, %STRUCT2 { i8 0, i32 2048 }, %STRUCT2 zeroinitializer, %STRUCT2 zeroinitializer, %STRUCT2 zeroinitializer] }, %STRUCT1 zeroinitializer, %STRUCT1 zeroinitializer, %STRUCT1 zeroinitializer] }
+@__STRUCT1__init = unnamed_addr constant %STRUCT1 zeroinitializer
+@__STRUCT2__init = unnamed_addr constant %STRUCT2 zeroinitializer
+@__mainProg.var_str1__init = unnamed_addr constant [5 x %STRUCT1] [%STRUCT1 { i16 1, [5 x %STRUCT2] [%STRUCT2 { i8 1, i32 128 }, %STRUCT2 { i8 0, i32 1024 }, %STRUCT2 zeroinitializer, %STRUCT2 zeroinitializer, %STRUCT2 zeroinitializer] }, %STRUCT1 { i16 2, [5 x %STRUCT2] [%STRUCT2 { i8 1, i32 256 }, %STRUCT2 { i8 0, i32 2048 }, %STRUCT2 zeroinitializer, %STRUCT2 zeroinitializer, %STRUCT2 zeroinitializer] }, %STRUCT1 zeroinitializer, %STRUCT1 zeroinitializer, %STRUCT1 zeroinitializer]
```

Comment:
I'm a bit confused whether or not this IR is 100% correct. My thinking was that anything program / pou-stateful related is directly written into it's instance (here `mainProg_instance`), yet there's also `__mainProg.var_str1__init` which isn't really used?

Reply:
the `__init` is the initializer that we generate for all structs. in the case of a `PROGRAM` this does not make much sense but is not a big deal for now. We could probably add an issue to remove the init. But this is a side effect of the program also being a struct.